### PR TITLE
Install repos on containerized hosts for python dependencies

### PR DESCRIPTION
**Pull request by sdodson**
- Removes dependency on openshift_facts
 - Adds Centos PaaS SIG common repo which is where origin users would get python2-ruamel-yaml
 - Makes lib_openshift and lib_utils depend on openshift_repos
 - Installs repos on containerized hosts for easier puddle testing

### CODE REVIEW

**Comment by sdodson** [comment]
Need to sort out python2-ruamel-yaml versus python-ruamel-yaml dependency. I noticed F25 python2-ruamel-yaml package doesn't provide python-ruamel-yaml.

File: roles/openshift_repos/tasks/main.yaml [hunk 1]
```diff
@@ -7,44 +7,51 @@
...
-        and not openshift.common.is_containerized | bool
-        and openshift_enable_origin_repo | default(true) | bool
+- name: openshift_repos detect ostree
+  stat:
+    path: /run/ostree-booted
+  register: ostree_booted
+
+- block:
+  - assert:
+      that: openshift_deployment_type in known_openshift_deployment_types
```
Comment (detiber) [inline roles/openshift_repos/tasks/main.yaml]: It almost seems like this assert should be outside of the block, even though the variable isn't used outside of the block.

The assert should probably have a `msg` defined (this wasn't available with Ansible 1.9).

File: roles/openshift_repos/tasks/main.yaml [hunk 2]
```diff
@@ -7,44 +7,51 @@
...
+
+  - name: Configure origin yum repositories RHEL/CentOS
+    copy:
+      src: origin/repos/openshift-ansible-centos-paas-sig.repo
+      dest: /etc/yum.repos.d/
+    notify: refresh cache
+    when:
+    - ansible_os_family == "RedHat" and ansible_distribution != "Fedora"
+    - openshift_deployment_type == 'origin'
+    - openshift_enable_origin_repo | default(true) | bool
```
Comment (detiber) [inline roles/openshift_repos/tasks/main.yaml]: I'd condense these two copy tasks into a single task:
```
- name: Configure origin gpg keys if needed
  copy:
    src: origin/gpg_keys/openshift-ansible-CentOS-SIG-PaaS
    dest: /etc/pki/rpm-gpg/
  notify: refresh cache
  with_items:
  - src: origin/gpg_keys/openshift-ansible-CentOS-SIG-PaaS
    dest: /etc/pki/rpm-gpg/
  - src: origin/repos/openshift-ansible-centos-paas-sig.repo
    dest: /etc/yum.repos.d/
  when:
  - ansible_os_family == "RedHat"
  - ansible_distribution != "Fedora"
  - openshift_deployment_type == 'origin'
  - openshift_enable_origin_repo | default(true) | bool
```

I'm not entirely sure that we need to keep `ansible_os_family == "RedHat"`, but it doesn't hurt to leave it in place.

File: roles/openshift_repos/tasks/main.yaml [hunk 3]
```diff
@@ -7,44 +7,51 @@
 #       subscription management module to take a poolid and enable/disable the
 #       proper repos correctly.
```
Comment (detiber) [inline roles/openshift_repos/tasks/main.yaml]: I think it's probably safe to delete the above TODO comments at this point

File: roles/openshift_repos/meta/main.yml [hunk 4]
```diff
@@ -12,4 +12,3 @@ galaxy_info:
   categories:
   - cloud
 dependencies:
```
Comment (detiber) [inline roles/openshift_repos/meta/main.yml]: This should be changed to: `dependencies: []`

File: roles/openshift_repos/tasks/main.yaml [hunk 5]
```diff
@@ -7,44 +7,51 @@
...
+    - openshift_additional_repos | length > 0
+    notify: refresh cache
+
+  - name: Remove the additional repos if no longer defined
+    file:
+      dest: /etc/yum.repos.d/openshift_additional.repo
+      state: absent
+    when:
+    - openshift_additional_repos | length == 0
+    notify: refresh cache
```
Comment (sdodson) [inline roles/openshift_repos/tasks/main.yaml]: state doesn't seem to be valid for templates.

File: roles/openshift_repos/tasks/main.yaml (hunk 3 above)
Comment (sdodson) [inline roles/openshift_repos/tasks/main.yaml]: updated

File: roles/openshift_repos/tasks/main.yaml (hunk 2 above)
Comment (sdodson) [inline roles/openshift_repos/tasks/main.yaml]: updated

File: roles/openshift_repos/meta/main.yml (hunk 4 above)
Comment (sdodson) [inline roles/openshift_repos/meta/main.yml]: updated

**Comment by sdodson** [comment]
aos-ci-test

File: roles/lib_utils/meta/main.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  license: Apache License, Version 2.0
+  min_ansible_version: 1.7
+  platforms:
+  - name: EL
+    versions:
+    - 7
+  categories:
+  - cloud
+dependencies:
+- { role: openshift_repos }
```
Comment (mtnbikenc) [inline roles/lib_utils/meta/main.yml]: roles/lib_utils/meta/main.yml:
  15:28     error    trailing spaces  (trailing-spaces)

File: roles/openshift_repos/tasks/main.yaml (hunk 5 above)
Comment (detiber) [inline roles/openshift_repos/tasks/main.yaml]: doh, of course it isn't.

File: roles/lib_utils/meta/main.yml (hunk 6 above)
Comment (sdodson) [inline roles/lib_utils/meta/main.yml]: thanks! fixed.

**Comment by sdodson** [comment]
aos-ci-test

**Pull request merged**